ceph/ceph_admin/iscsi.py

```diff
@@ -1,77 +1,50 @@
-        Args:
-            config: apply config data
+        Deploy ISCSI client daemon using the provided arguments.
 
-        config:
-            nodes: list of nodes where iscsi to be deployed
-            pool: pool_name
-            api_user: username
-            api_password: user_password
-            trusted_ip_list: [ip_addr1, ip_addr2]
+        Args:
+            config: Key/value pairs provided from the test scenario
+
+        Example:
+            config:
+                command: apply
+                service: iscsi
+                base_cmd_args:          # arguments to ceph orch
+                    concise: true
+                    verbose: true
+                    input_file: <name of spec>
+                pos_args:
+                    - india             # name of the pool
+                    - api_user          # name of the API user
+                    - api_pass          # password of the api_user.
+                    - trusted_ip_list   # space separate list of IPs
+                args:
+                    placement:
+                        label: iscsi    # either label or node.
+                        nodes:
+                            - node1
+                        limit: 3    # no of daemons
+                        sep: " "    # separator to be used for placements
+                    dry-run: true
+                    unmanaged: true
         """
-        pool = config.get("pool_name", "iscsi")
-        pg_num = config.get("pool_pg_num", 3)
-        pgp_num = config.get("pool_pgp_num", 3)
-        user = config.get("api_user", "user")
-        password = config.get("api_password", "password")
-        trusted_ip_list = config.get("trusted_ip_list", [])
-
-        cmd = Apply.apply_cmd + [self.__ROLE]
-        nodes = get_nodes_by_id(self.cluster, config.get("nodes"))
-
-        cmd.extend([pool, user, password, repr(" ".join(trusted_ip_list))])
-
-        if not nodes:
-            raise ResourcesNotFoundError("Nodes not found: %s", config.get("nodes"))
-
-        host_placement = [node.shortname for node in nodes]
-        cmd.append("--placement '{}'".format(";".join(host_placement)))
-
-        # Create pool
-        self.shell(
-            remote=self.installer,
-            args=[
-                "ceph",
-                "osd",
-                "pool",
-                "create",
-                pool,
-                str(pg_num),
-                str(pgp_num),
-                "replicated",
-            ],
-        )
-
-        # Associate pool to RBD application
-        self.shell(
-            remote=self.installer,
-            args=["ceph", "osd", "pool", "application", "enable", pool, "rbd"],
-        )
+        pos_args = config.pop("pos_args")
 
-        Apply.apply(
-            self,
-            role=self.__ROLE,
-            command=cmd,
-            placements=host_placement,
-        )
+        if isinstance(pos_args[-1], list):
```

Comment:
Do we need to make this dynamic? That is pull the `CephNode` for the node names and then use the associated IP address.

Maybe we can take it up as an RFE.